app/controllers/package_set.js

```diff
@@ -36,27 +38,30 @@ export default Controller.extend({
   allPackages: computed(
     "model",
     "model.isSet",
-    "model.packagesAndSets.@each.isAvailable",
+    "model.packagesAndSets@each.isAvailable",
     function() {
       var record = this.get("model");
       if (!record) {
         this.send("noItemsPresent");
+        this.set("packageUnavailableInSet", true);
         return [];
       }
 
-      return record.get("isSet")
-        ? record.get("packages").filterBy("isAvailable")
-        : [record];
+      return record.get("isSet") ? record.get("packages") : [record];
     }
   ),
 
-  notAvailableInStock: computed(
+  notAvailableInStock: observer(
     "allPackages.@each.availableQuantity",
     function() {
-      let quantities = this.get("allPackages").map(pkg =>
-        pkg.get("availableQuantity")
+      let quantity = this.get("allPackages").any(
```

Comment:
shouldn't this be `.some()`? 🤔

Reply:
sure..even that can be done..i will do it